api/v1/nodeptpdevice_types.go

```diff
@@ -21,47 +21,73 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+//+kubebuilder:printcolumn:name="HwConfig",type="string",JSONPath=".status.hwconfig.status"
+
+// NodePtpDevice is the Schema for the nodeptpdevices API
+type NodePtpDevice struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NodePtpDeviceSpec   `json:"spec,omitempty"`
+	Status NodePtpDeviceStatus `json:"status,omitempty"`
+}
 
 // NodePtpDeviceSpec defines the desired state of NodePtpDevice
 type NodePtpDeviceSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
 }
 
 type PtpDevice struct {
-	Name    string `json:"name,omitempty"`
+	// Name is the name of the PTP device.
+	// It is a unique identifier for the device.
+	// +required
```

Comment:
my understanding is anything `omitempty` should be optional. In this case I think the Name should be required, and we should remove `omitempty`. @aneeshkp ?